### PR TITLE
Fix auto cursor placement for keyboard input for datepicker

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #181 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

When the user has a valid date in the datepicker input and then exclusively uses the keyboard to change the date, the cursor placed by the user now stays in place, rather than automatically moving to the end of the date input string.

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: src/base-input.js [hunk 1]
```diff
@@ -442,6 +453,14 @@ export default class BaseInput extends LitElement {
...
     const { selectionStart } = this.inputElement;
 
+    this.inputElement.addEventListener('keyup', (evt) => {
+      if (evt.key === 'Backspace') {
+        if (this.type === 'month-day-year' && this.value) {
+          this.inputElement.setSelectionRange(selectionStart, selectionStart);
+        }
+      }
+    });
+
```
[the remark this answers is not in the document]
Comment (blackfalcon) [reply src/base-input.js:463]: I see that you have addressed the issue with the initial cursor placement and this does allow the user to change that one value. Then we run into that issue with Cleave trying to format the data.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: src/base-input.js (hunk 1 above)
Comment (jason-capsule42) [inline src/base-input.js:463]: This isn't handling when CleaveJS is auto-formatting the input. I have an idea. Let's do some pair-programming today.

**Review by jason-capsule42 — CHANGES_REQUESTED** [review]
Needs to handle Autoformatting from CleaveJS

**Review by blackfalcon — DISMISSED** [review]
I am typing in the following date `4121973`

Sep-21-2022 10-25-11

**Review by blackfalcon — APPROVED** [review]
This update is 99.999% successful. The only remaining edge case I discovered is if you were to put in a date like `04/12/1988` and then to change the `04` to `12` the ONLY way to make that happen is to highlight the `04` and then type `12`. Anything else causes chaos between cursor placement and auto-formatting. 

But this is an edge case. If this were to come up as customer feedback, then we can talk about the nuclear option.

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**